components/brave_vpn/resources/panel/state/async.ts

```diff
@@ -50,30 +60,40 @@ handler.on(Actions.purchaseConfirmed.getType(), async (store) => {
   const [{ state }, { currentRegion }, { regions }] = await Promise.all([
     getPanelBrowserAPI().serviceHandler.getConnectionState(),
     getPanelBrowserAPI().serviceHandler.getSelectedRegion(),
-    getPanelBrowserAPI().serviceHandler.getAllRegions()
+    getPanelBrowserAPI().serviceHandler.getAllRegions(
+      REGION_PRECISION_CITY_BY_COUNTRY
+    )
   ])
 
-  store.dispatch(Actions.showMainView({
-    currentRegion,
-    regions,
-    connectionStatus: ((state === ConnectionState.CONNECT_FAILED)
-    ? ConnectionState.DISCONNECTED : state), /* Treat connection failure on startup as disconnected */
-    expired: false
-  }))
+  store.dispatch(
+    Actions.showMainView({
+      currentRegion,
+      regions,
+      connectionStatus:
+        state === ConnectionState.CONNECT_FAILED
+          ? ConnectionState.DISCONNECTED
+          : state /* Treat connection failure on startup as disconnected */,
+      expired: false
+    })
+  )
 })
 
 handler.on(Actions.purchaseExpired.getType(), async (store) => {
   const [{ currentRegion }, { regions }] = await Promise.all([
     getPanelBrowserAPI().serviceHandler.getSelectedRegion(),
-    getPanelBrowserAPI().serviceHandler.getAllRegions()
+    getPanelBrowserAPI().serviceHandler.getAllRegions(
+      REGION_PRECISION_CITY_BY_COUNTRY
```

Comment:
Do we ever call this function with a different precision?

Reply:
I assumed that it may be useful for the desktop as it provides a different response based on the region_precision. If it's not useful for the desktop, we can surely get rid of it.

Reply:
Fixed by https://github.com/brave/brave-core/pull/25089/commits/6f24b64ae3ffd988ca734d5557fedc02aea6c771